pkg/handler/handler.go

```diff
@@ -12,19 +13,53 @@ import (
 	"time"
 
 	"cloud.google.com/go/storage"
+
+	"golang.org/x/oauth2"
+	googleOAuth2 "golang.org/x/oauth2/google"
+
+	"github.com/dghubble/gologin"
+	"github.com/dghubble/gologin/google"
+	"github.com/dghubble/sessions"
+)
+
+// TODO: update these values
+const (
+	sessionName    = "example-google-app"
+	sessionSecret  = "example cookie signing secret"
+	sessionUserKey = "googleID"
 )
 
 var pathChecker = regexp.MustCompile("logs/[a-zA-Z_-]+/[0-9]+/[ a-zA-Z_-]+/[0-9]+")
 
+// sessionStore encodes and decodes session data stored in signed cookies
+var sessionStore = sessions.NewCookieStore([]byte(sessionSecret), nil)
+
 type prowBucketHandler struct {
-	bucket *storage.BucketHandle
-	tmpDir string
+	bucket       *storage.BucketHandle
+	tmpDir       string
+	clientID     string
+	clientSecret string
+	requestURL   string
 }
 
-func New(b *storage.BucketHandle, cacheDir, listenAddress string) *http.Server {
-	bucketHandler := &prowBucketHandler{bucket: b, tmpDir: cacheDir}
+func New(b *storage.BucketHandle, cacheDir, listenAddress, clientID, clientSecret string) *http.Server {
+	bucketHandler := &prowBucketHandler{bucket: b, tmpDir: cacheDir, clientID: clientID, clientSecret: clientSecret}
 	mux := &http.ServeMux{}
-	mux.HandleFunc("/", bucketHandler.router)
+	mux.HandleFunc("/", bucketHandler.welcomeHandler)
+	mux.HandleFunc("/logout", bucketHandler.logoutHandler)
+
+	oauth2Config := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		RedirectURL:  "http://localhost:8080/google/callback", // TODO: change this!
+		Endpoint:     googleOAuth2.Endpoint,
+		Scopes:       []string{"email"},
```

Comment:
Is some sort of authz implied in the `ClientID` and `ClientSecret` or does this mean anyone with a valid google account can login?

Reply:
> does this mean anyone with a valid google account can login?

It means this. :+1: